core/src/main/java/bio/terra/pearl/core/service/export/EnrolleeExportService.java

```diff
@@ -296,5 +299,40 @@ protected BaseExporter getExporter(ExportFileFormat fileFormat, List<ModuleForma
         return new TsvExporter(moduleFormatters, enrolleeMaps, fileFormat, columnSorting);
     }
 
+    private Map<UUID, List<SurveyResponseWithTaskDto>> attachTasksToSurveyResponses(Map<UUID, List<ParticipantTask>> taskMap, Map<UUID, List<SurveyResponse>> surveyResponseMap) {
+        Map<UUID, List<SurveyResponseWithTaskDto>> surveyResponseWithTaskMap = new HashMap<>();
+
+        for (UUID enrolleeId : surveyResponseMap.keySet()) {
+            List<ParticipantTask> tasks = taskMap.get(enrolleeId);
+            List<SurveyResponse> surveyResponses = surveyResponseMap.get(enrolleeId);
+            List<SurveyResponseWithTaskDto> surveyResponseWithTaskDtos = new ArrayList<>();
+
+            if (tasks == null) {
+                tasks = Collections.emptyList();
+            }
+
+            for (SurveyResponse surveyResponse : surveyResponses) {
+                SurveyResponseWithTaskDto surveyResponseWithTaskDto = new SurveyResponseWithTaskDto();
+
+                try {
+                    BeanUtils.copyProperties(surveyResponseWithTaskDto, surveyResponse);
+                } catch (Exception e) {
+                    throw new IllegalStateException("Error copying properties from surveyResponse to surveyResponseWithTaskDto", e);
+                }
```

Comment:
How about making a SurveyResponseWithTaskDto constructor that takes a SurveyResponse and a ParticipantTask as arguments?  That will make the split between task-finding and dto assembly clearer